packages/synapse-react-client/src/components/LoadingScreen/index.ts

```diff
@@ -0,0 +1,2 @@
+import { SynapseSpinner, BlockingLoader } from './LoadingScreen'
+export { SynapseSpinner, BlockingLoader }
```

Comment:
Should show some kind of loading while the command is executing, but I'm not sure how we would do this today given the current architecture.  Any ideas, @nickgros ?

Reply:
This is an interesting problem. Since this is just dealing with an external data source, my gut is telling me that we should just somehow use react-query since that will let us neatly handle loading/error cases.

Is the loading behavior generalizable to all of the custom buttons? Maybe instead of the synchronous `onClick` prop, we just pass in arguments that match the arguments to `useMutation`? So we would have 

* a `mutationFn` prop, where the passed prop would be like:
```
async (e: CustomControlCallbackData) => {
  await SynapseClient.getFullQueryTableResults(/* ... */ );
  localStorage.setItem(/* ... */)
  return
}
```
* an `onSuccess` prop, where the passed prop would be like:
```
() => {
  window.location.href = '...'
}
```

Optionally have an `onError` prop, but I'm not sure what to do in this specific case.

The button component itself then manages loading state based on the return value of `useMutation`.

If we wanted different loading behavior, that could be an enum/boolean prop too I guess?

I feel like this idea might be overcomplicated.